lock-impl/src/test/java/com/palantir/lock/client/LockRefreshingLockServiceTest.java

```diff
@@ -74,4 +79,19 @@ public void testClosed() {
                 .isInstanceOf(SafeIllegalStateException.class)
                 .hasMessage("LockRefreshingLockService is closed");
     }
+
+    @Test
+    public void testFailedRefreshCallback() throws InterruptedException {
+        AtomicReference<Set<LockRefreshToken>> failedTokens = new AtomicReference<>();
+        server.registerRefreshFailedCallback(failedTokens::set);
+        LockRequest request = LockRequest.builder(ImmutableSortedMap.of(lock1, LockMode.WRITE))
+                .timeoutAfter(SimpleTimeDuration.of(5, TimeUnit.SECONDS))
+                .build();
+        LockResponse lock = server.lockWithFullLockResponse(LockClient.ANONYMOUS, request);
+        // use the internal server to unlock (refreshing service will still try to refresh but fail)
+        internalServer.unlock(lock.getToken());
+        Thread.sleep(10000);
```

Comment:
Can we use awaitility instead of sleeping?

Reply:
beyond this, please can we figure out a way to have tests that don't just take 10 seconds? seems pretty unreasonable

Reply:
Unfortunately the 5-second refresh time is hard-coded in LockRefreshingLockService and all of the refresh logic is not exposed in any way that tests can interact with it. We'd need to refactor that to be able to make tests shorter. I think refactoring LockRefreshingLockService is out of scope for the change I'm trying to make.